src/test/java/com/epam/ta/tests/PostsTest.java

```diff
@@ -1,45 +1,141 @@
+        assertThat(posts.size()).isEqualTo(100);
+
+    }
+
+    @Test
+    @DisplayName("Get one Post")
+    public void verifyPost() {
+        var response = PostEndpoint.getPost(TEST_POST_ID);
+
+        assertThat(response.getStatusCode())
+                .as("wrong status code")
+                .isEqualTo(HttpStatus.SC_OK);
+
+        var post = new Gson().fromJson(response.jsonPath().prettify(), Post.class);
+        PostSteps.verifyPost(post, TEST_POST_ID, TEST_USER_ID);
+    }
+
+    @Test
+    @DisplayName("Get all comments for Post")
+    public void verifyPostComments() {
+        var response = PostEndpoint.getPostComments(TEST_POST_ID);
+
+        assertThat(response.getStatusCode())
+                .as("wrong status code")
+                .isEqualTo(HttpStatus.SC_OK);
+
+        var comments = Arrays.asList(new Gson().fromJson(response.jsonPath().prettify(), Comment[].class));
+        assertThat(comments.size()).isEqualTo(5);
+    }
+
+    @Test
+    @DisplayName("Get all Posts filter by UserId")
+    public void verifyPostsFilterByUserId() {
+        var response = PostEndpoint.getPostsFilterByUserId(TEST_USER_ID);
+
+        assertThat(response.getStatusCode())
+                .as("wrong status code")
+                .isEqualTo(HttpStatus.SC_OK);
+
+        var posts = Arrays.asList(new Gson().fromJson(response.jsonPath().prettify(), Post[].class));
+        assertThat(posts.size()).isEqualTo(10);
     }
 
     @Test
-    public void checkStatusCode() {
-        Response response = RestAssured.when()
-                .get("/posts")
-                .andReturn();
-        Assert.assertEquals(response.getStatusCode(), 200);
+    @DisplayName("Add new Post")
+    public void verifyAddPost() {
+        var testTitle = "title#1";
+        var testBody = "body#1";
+
+        var response = PostEndpoint.addPost(TEST_USER_ID, testTitle, testBody);
+
+        assertThat(response.getStatusCode())
+                .as("wrong status code")
+                .isEqualTo(HttpStatus.SC_CREATED);
+
+        var post = new Gson().fromJson(response.jsonPath().prettify(), Post.class);
+        PostSteps.verifyPost(post, 101, TEST_USER_ID, testTitle, testBody);
     }
 
     @Test
-    public void checkResponseHeader() {
-        Response response = RestAssured.when()
-                .get("/posts")
-                .andReturn();
-        String rpContentTypeHeader = response.getHeader("Content-Type");
-        Assert.assertEquals(rpContentTypeHeader, "application/json; charset=utf-8");
+    @DisplayName("Delete existing Post")
+    public void verifyDeletePost() {
+        var response = PostEndpoint.deletePost(TEST_POST_ID);
+
+        assertThat(response.getStatusCode())
+                .as("wrong status code")
+                .isEqualTo(HttpStatus.SC_OK);
+
+        //Should be TEST_POST_ID as parameter but was set as non existing postId just to simulate deletion process
+        var nullResponse = PostEndpoint.getPost(123);
```

Comment:
i think you don't need that at all if Api does not return anything and it's ok :)